include/nil/actor/zk/snark/systems/plonk/placeholder/verifier.hpp

```diff
@@ -88,13 +88,10 @@ namespace nil {
 
                         for (std::size_t i = 0; i < permutation_size; i++) {
                             std::size_t zero_index = 0;
-                            for (std::size_t j = 0;
-                                 j < preprocessed_public_data.common_data.columns_rotations[i].size();
-                                 j++) {
-
-                                if (preprocessed_public_data.common_data.columns_rotations[i][j] == 0) {
-                                    zero_index = j;
-                                }
+                            for (int v: preprocessed_public_data.common_data.columns_rotations[i]) {
```

Comment:
We need to do some refactoring `zk` library I guess...

```
std::size_t zero_index = std::find(preprocessed_public_data.common_data.columns_rotations[i].begin(), preprocessed_public_data.common_data.columns_rotations[i].end(), 0)
```

Reply:
That's true, but let's do it later. I'm planning to read the placeholder code and try to refactor/optimize it in a few weeks. There are plenty of such small things, and I'm removing them over time.